### DISCUSSION TITLE
Add example: Getting UV projected texture image

### DISCUSSION THREAD

**Discussion by Karric** [marcofugaro/three-projected-material#53] [Q&A]
Please add example code of how I can get the mesh's UV texture with the projected image applied to it.

Please also consider the scenario where the mesh is not unwrapped and thus has no base texture.

**Comment by marcofugaro** [comment]
not sure what you mean by "get the mesh's UV texture", can you elaborate what you're trying to do?

**Reply by Karric** [reply]
Let's say I have an image of a 6-sided die. I load a cube and rotate it to match the image, then use your class to project the image onto the cube. I now want to export the cube's new texture (of the projected image) for use in another program as a traditional UV texture.

**Reply by Karric** [reply]
@marcofugaro is this possible? I looked at your class and I do not know shaders/webgl well enough to add this myself.

**Reply by marcofugaro** [reply]
What you're talking about is texture baking,  it is possible in three.js, but you'd have to do it by hand.
There is not example in three.js, but you might find some resources in the issue: https://github.com/mrdoob/three.js/issues/14055.

Not related to this library, you could bake any three.js material.

**Reply by Karric** [reply]
@marcofugaro ok, understood. I'm having a second issue. It seems like ProjectedMaterial requires lights in the scene. Is there some way to render this material as shadeless? I tried passing flatShading: THREE.FlatShading when I create the material, but this did not do anything. I added an ambient light, but it lightens the projected texture too much. I checked online and could not find very much about shadeless.

**Reply by monschine** [reply]
Hi Marco, as Karric says, don't you think it could be possible to uv unwrap the texture once projected to the mesh ? so let's say a second shader that reads the current pixels projected on the mesh and fit the uvmaps that the object already has. So let's say an scenario where you load an obj file with uv maps (created in blender or any other software) and then projecting the pixels form the initial texture projection to the final uv maps projection ?
Thanks in advance

**Reply by monschine** [reply]
Here I prepared an illustration to show visually what could be achieved directly via shader 
![texture_baking_openGL](https://github.com/marcofugaro/three-projected-material/assets/9587968/ef40f9f4-2e41-4c1b-a5e7-10a98611f3dc)

**Reply by marcofugaro** [reply]
Yes it's possible, that's what texture baking does. But you have to do it yourself because it's out of scope of this library.
